app/questionnaire/hub.py

```diff
@@ -0,0 +1,136 @@
+from flask_babel import lazy_gettext as _
```

Comment:
It kind of feels like this should be two classes. One that calculates and stores the overall state of the survey, and another that is a view context that takes that and does the view specific mapping. It would also mean the survey progress would be more unit testable without getting into view specifics.

Reply:
`ProgressStore` now stores the state of the survey/section and the `Hub` is then initialized using `ProgressStore` to build the view context.